components/test-suite/src/loci/tests/testng/FormatReaderTestFactory.java

```diff
@@ -199,29 +202,32 @@ public String getProperty(String key) {
     }
 
     // remove duplicates
-    int index = 0;
+    Set<String> fileSet = new HashSet<String>(files);
```

Comment:
The main downside of this operation is that it works under the assumption that the master/current file selection independently of the initially selected file in `setId()`. Unfortunately, this does not hold for many multi-file readers which do not implement such logic (incl. `ome-tiff`, `micromanager`...).

Since the file header is expected to match the output of `getCurrentFile()` for the first scanned file of the fileset, is there a way to maintain the previous listing order?